Enformer.pm

```diff
@@ -0,0 +1,156 @@
+
+ This Plugin is available for GRCh37 and GRCh38
+ 
+ Please cite the Enformer publication alongside the VEP if you use this resource:
+ https://www.nature.com/articles/s41592-021-01252-x
+
+ Enformer scores can be downloaded from https://ftp.ensembl.org/pub/current_variation/Enformer for GRCh37 and GRCh38.
+
+ The plugin can then be run as default:
+ ./vep -i variations.vcf --assembly GRCh38 --plugin Enformer,file=/path/to/Enformer/data.vcf.gz
+
+ or run with option to only retrieve the SAD score 
+ ./vep -i variations.vcf --assembly GRCh38 --plugin Enformer,file=/path/to/Enformer/data.vcf.gz,SAD=1 
+ 
+ or run with option to only retrieve the SAR score 
+ ./vep -i variations.vcf --assembly GRCh38 --plugin Enformer,file=/path/to/Enformer/data.vcf.gz,SAR=1 
+
+
+ The tabix utility must be installed in your path to use this plugin.
+ Check https://github.com/samtools/htslib.git for instructions.
+
+
+=cut
+
+package Enformer;
+
+use strict;
+use warnings; 
+
+use Bio::EnsEMBL::Variation::Utils::BaseVepTabixPlugin;
+use base qw(Bio::EnsEMBL::Variation::Utils::BaseVepTabixPlugin);
+
+sub new {
+  my $class = shift;
+  
+  my $self = $class->SUPER::new(@_);
+
+  $self->expand_left(0);
+  $self->expand_right(0);
+  $self->get_user_params();
+  
+  my $params = $self->params_to_hash();
+  
+  my $file;
+  if (!keys %$params) {
+    $file = $self->params->[0];
+    $params->{file} = $file;
+  } else {
+    $file = $params->{file};
+    $self->{SAD} = $params->{SAD} if (defined ($params->{SAD}));
+    $self->{SAR} = $params->{SAR} if (defined ($params->{SAR}));
+  } 
+  
+  $self->{params} = $params;
+  $self->add_file($file);
+  
+  my $assembly = $self->{config}->{assembly};
+
+  return $self;
+
+}
+
+sub feature_types {
+  return ['Transcript','RegulatoryFeature','MotifFeature','Intergenic'];
+}
+
+sub get_header_info {
+  return { Enformer => "Prediction tool to accurately identify variant impact on gene expression" };
+}
+
+sub run{
+  my ($self, $tva) = @_;
+  
+  my $vf = $tva->variation_feature;
+  my $allele = $tva->variation_feature_seq;
+  return {} unless $allele =~ /^[ACGT]$/;
+  
+
+  my ($vf_start, $vf_end) = ($vf->{start}, $vf->{end});
+  ($vf_start, $vf_end) = ($vf_end, $vf_start) if ($vf_start > $vf_end);
+
+  my ($res) = grep{
+    $_->{alt} eq $allele &&
+    $_->{start} == $vf_start &&
+    $_->{start} == $vf_end
+    } @{$self->get_data($vf->{chr}, $vf_start, $vf_end)};
+    
+  
+  return $res ? $res->{result} : {} if keys($self->{params}) == 1;
+
+  return $res ? {Enformer => $res->{SAD}} : {} if defined($self->{SAD});
+
+  return $res ? {Enformer => $res->{SAR}} : {} if defined($self->{SAR});
+}
+
+
+sub parse_data {
+  my ($self, $line) = @_;
+  my ($chr_data, $start, $snp, $ref, $alt, $qual, $filter, $data) = split("\t", $line);
+  
+  my ($chr) = $chr_data =~ /(\d+)/; # this is because the chromosome is chr1 etc, to retrieve just the 1
```

Comment:
This regex doesn't work with `chrX` (and potentially others like `chrY` and `chrMT`, if they are added in the future).
```suggestion
  my ($chr) = $chr_data =~ /chr(.+)/; # this is because the chromosome is chr1 etc, to retrieve just the 1
```